ios/MarkdownCommitHook.mm

```diff
@@ -0,0 +1,142 @@
+
+#include "MarkdownCommitHook.h"
+#include "RCTMarkdownStyle.h"
+#include "RCTMarkdownUtils.h"
+#include "MarkdownShadowFamilyRegistry.h"
+
+using namespace facebook::react;
+
+namespace livemarkdown {
+
+MarkdownCommitHook::MarkdownCommitHook(const std::shared_ptr<UIManager> &uiManager) : uiManager_(uiManager) {
+  uiManager_->registerCommitHook(*this);
+}
+
+MarkdownCommitHook::~MarkdownCommitHook() noexcept {
+  uiManager_->unregisterCommitHook(*this);
+}
+
+RootShadowNode::Unshared MarkdownCommitHook::shadowTreeWillCommit(
+    ShadowTree const &,
+    RootShadowNode::Shared const &,
+    RootShadowNode::Unshared const &newRootShadowNode) noexcept {
+        auto rootNode = newRootShadowNode->ShadowNode::clone(ShadowNodeFragment{});
+
+        // A preface to why we do the weird thing below:
+        // On the new architecture there are two ways of measuring text on iOS: by value and by pointer.
+        // When done by value, the attributed string to be measured is created on the c++ side. We cannot
+        // modify this process as we do not extend TextInputShadowNode. We also cannot really change the
+        // layout manager used to do this, since it's a private field (ok, we can but in a not very nice way).
+        // But also, the logic for parsing and applying markdown is written in JS/ObjC and we really wouldn't
+        // want to reimplement it in c++.
+        //
+        // Nice thing is that it can also be done by pointer to NSAttributedString, which is the platform's
+        // way to handle styled text, and is also used by Live Markdown. On this path, the measurement is done
+        // by the OS APIs. The thing we want to make sure of, is that markdown-decorated text input always uses
+        // this path and uses a pointer to a string with markdown styles applied. Thankfully, RN provides nice
+        // utility functions that allow to convert between the RN's AttributedString and iOS's NSAttributedString.
+        // The logic below does exactly that.
+        
+        // In order to properly apply markdown formatting to the text input, we need to update the TextInputShadowNode's
+        // state with styled string, but we only have access to the ShadowNodeFamilies of the decorator components.
+        // We also know that a markdown decorator is always preceded with the TextInput to decorate, so we need to take
+        // the sibling.
+        std::vector<MarkdownTextInputDecoratorPair> nodesToUpdate;
+        MarkdownShadowFamilyRegistry::runForEveryFamily([&rootNode, &nodesToUpdate](ShadowNodeFamily::Shared family) {
+         // get the path from the root to the node from the decorator family
+         auto ancestors = family->getAncestors(*rootNode);
+         
+         if (!ancestors.empty()) {
+             auto &parentNode = ancestors.back().first.get();
+             auto index = ancestors.back().second;
+             
+             auto markdownNode = parentNode.getChildren().at(index);
```

Comment:
Renamed it to `decoratorNode` in https://github.com/Expensify/react-native-live-markdown/pull/199/commits/4e4f30541cadea53a0a4c4a4e7c820c4c2d7a09c to be explicit which node it refers to since below `texInputNode` is declared. WDYT?